### PR TITLE
Add Pytorch Native AMP support in Trainer

### LINKED ISSUE #6115: Usage of Pytorch Native AMP in place of apex (Pytorch 1.6) in Trainer
# 🚀 Feature request
It would be nice to remove the Apex dependency for `fp16` training and use the native [Pytorch's AMP methods](https://github.com/pytorch/pytorch/releases) in the `Trainer` method. Pytorch recommends Apex users to switch to it's native implementation, even [Apex does it so](https://github.com/NVIDIA/apex/issues/818). Moreover, it will eliminate the need for users to build apex by themselves. 

## Your contribution
I am happy to submit a PR if you think it would be a good addition. Please let me know.

### PR DESCRIPTION
**Pull request by prajjwal1**
Pytorch 1.6 introduces Native AMP support. This eliminates the need to build and install Apex and provides improvements over problems highlighted in [Apex #818](https://github.com/NVIDIA/apex/issues/818) and flexibility. This is the recommended way to use AMP.

With this PR, Trainer will automatically use Pytorch's native AMP if 1.6 version is installed, otherwise, it will use Apex.

This PR will close [#6115](https://github.com/huggingface/transformers/issues/6115).

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -33,8 +33,19 @@
 from .training_args import TrainingArguments
 
 
-if is_apex_available():
-    from apex import amp
+_use_native_amp = False
+_use_apex = False
+
+# Check if Pytorch version >= 1.6 to switch between Native AMP and Apex
+if not version.parse(torch.__version__) >= version.parse("1.6"):
```
Comment (sgugger) [inline src/transformers/trainer.py:40]: Nit: can we remove the not and either use < in the comparison or switch the two parts inside the test? I find it hard to read this way.

**Review by sgugger — APPROVED** [review]
Thanks a lot for the PR! It's very clean, I just have one suggestion.

**Review by julien-c — APPROVED** [review]

File: src/transformers/trainer.py (hunk 1 above)
Comment (prajjwal1) [reply src/transformers/trainer.py]: Thanks for the suggestion. This is resolved now.

**Review by LysandreJik — APPROVED** [review]
That's really cool, thanks a lot @prajjwal1 !

**Pull request merged**